### DISCUSSION TITLE
文件选择框&路径框怎么搞?

### DISCUSSION THREAD

**Discussion by pkptzx** [syf20020816/SurrealismUI#6] [Q&A] [answered]

**Comment by syf20020816** [comment, marked as answer]
# About Question

这是一个非常好的想法，在GUI中文件选择器是十分重要的，但SurrealismUI是作为组件层存在，不直接操控Rust|C++进行操作，所以无法在SurrealismUI中存在直接点击按钮能够加载目标文件目录资源的文件选择器

This is a very good idea. File selectors are very important in the GUI, but SurrealismUI exists as a component layer and does not directly manipulate Rust | C++for operations. Therefore, there is no file selector in SurrealismUI that can directly click the button to load the target file directory resources

## Extra Component 

虽然SurrealismUI无法帮助使用者直接加载系统文件目录，但将在后续版本中支持树型控件(Tree)，其目的是为了以树的方式展现数据

Although SurrealismUI cannot assist users in directly loading system file directories, it will support Tree controls in future versions to present data in a tree like manner

### Target Support Version 

SurrealismUI V0.2.1 (after about 1 week)

## How To Solve Now

你可以使用SurrealismUI自定义组件，将输入属性设计为树型输入的结构，通过使用Rust|C++相关fs的API读取系统文件层级并转换为所需的输入

You can use SurrealismUI custom components to design input attributes as a tree like input structure. By using the API of Rust | C++related fs, you can read the system file hierarchy and convert it into the required input

## reference project

see : https://codeberg.org/flovansl/co_sl/src/branch/main/apps/coop_local

**Comment by syf20020816** [comment]
Now you can use `SURFile` or `SURTree`  in SurrealismUI V0.2.1
